### PR TITLE
user layers are implemented

### PR DESCRIPTION
**Pull request by andromash**

### CODE REVIEW

File: src/main/java/com/hibernate/cinema/Main.java [hunk 1]
```diff
@@ -37,5 +40,26 @@ public static void main(String[] args) {
         movieSessionService.add(movieSession);
         movieSessionService.findAvailableSessions(movie.getId(),
                 LocalDate.now()).forEach(System.out::println);
+
+        AuthenticationService authenticationService
+                = (AuthenticationService) injector.getInstance(AuthenticationService.class);
+        User andrii = authenticationService
+                .register("andriiromash@gmail.com", "abrakadabra");
+        User bob = authenticationService
+                .register("bobinpop@gmail.com", "bobbystrawbby");
```
Comment (Rommelua) [inline src/main/java/com/hibernate/cinema/Main.java:49]: Where is Alisa?? Poor Bob alone...

**Comment by Rommelua** [comment]
Good

File: src/main/java/com/hibernate/cinema/service/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
+        return userService.add(user);
+    }
+
+    private boolean isNotValid(String password, User userFromDB) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/hibernate/cinema/service/impl/AuthenticationServiceImpl.java]:
```suggestion
    private boolean isPasswordNotValid(String password, User userFromDB) {
```

File: src/main/java/com/hibernate/cinema/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        byte[] salt = HashUtil.getSalt();
```
Comment (RomanDubovskyi) [inline src/main/java/com/hibernate/cinema/service/impl/UserServiceImpl.java]: you don't need separate variable for salt, set it to user right away

File: src/main/java/com/hibernate/cinema/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        String password = HashUtil.hashPassword(user.getPassword(), salt);
```
Comment (RomanDubovskyi) [inline src/main/java/com/hibernate/cinema/service/impl/UserServiceImpl.java]:
```suggestion
        String hashedPassword = HashUtil.hashPassword(user.getPassword(), salt);
```

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**